src/static/bitbucket/utils/getWidgetSettings.ts

```diff
@@ -1,10 +1,25 @@
 import { WidgetSettings } from '../../../types';
+import { proxyRequestBare } from './RequestProxy';
+import useState from 'react-usestateref';
+import { useEffect } from 'react';
 
-export default function getWidgetSettings(): WidgetSettings {
-  return {
-    refreshInterval: 10000,
-    refreshOnlyWhenInViewport: false,
-    // @ts-ignore -- this is a global variable that may be set by the server
-    ...window.__WIDGET_SETTINGS__,
-  };
+const defaultSettings = {
+  refreshInterval: 10000,
+  refreshOnlyWhenInViewport: false,
+};
+
+export default function useWidgetSettings(): WidgetSettings {
+  const [widgetSettings, setWidgetSettings] = useState<WidgetSettings>(defaultSettings);
+
+  useEffect(() => {
+    proxyRequestBare<any>('/settings', 'POST')
```

Comment:
any reason to not type the API response?

Reply:
Nah, I should — copy pasta from API calls that existed before.